src/main/java/de/flo56958/minetinker/modifiers/ModManager.java

```diff
@@ -1,1137 +1,1170 @@
+	private String ArmorIdentifier;
+	private List<String> loreScheme;
+	private String modifierLayout;
+
+	private boolean allowBookConvert;
+
+	/**
+	 * Class constructor (no parameters)
+	 */
+	private ModManager() {
+	}
+
+	/**
+	 * get the instance that contains the modifier list (VERY IMPORTANT)
+	 *
+	 * @return the instance
+	 */
+	public static @NotNull ModManager instance() {
+		synchronized (ModManager.class) {
+			if (instance == null) {
+				instance = new ModManager();
+				instance.init();
+			}
+		}
+
+		return instance;
+	}
+
+	public static @Nullable Pair<@Nullable Material, @NotNull Integer> itemUpgrader(@NotNull Material tool, Material material) {
+		String name = tool.name().split("_")[0].toLowerCase();
+
+		if (name.equals("wooden") && material.name().contains("PLANKS")
+				|| name.equals("stone") && material == Material.COBBLESTONE
+				|| name.equals("iron") && material == Material.IRON_INGOT
+				|| name.equals("gold") && material == Material.GOLD_INGOT
+				|| name.equals("diamond") && material == Material.DIAMOND
+				|| name.equals("leather") && material == Material.LEATHER
+				|| name.equals("turtle") && material == Material.SCUTE
+				|| name.equals("netherite") && material == Material.NETHERITE_INGOT
+				|| name.equals("chainmail") && material == Material.IRON_BARS) {
+			return null;
+		}
+
+		if (ToolType.SWORD.contains(tool)) {
+			return new Pair<>(getToolUpgrade(material, "SWORD"), 2);
+		} else if (ToolType.PICKAXE.contains(tool)) {
+			return new Pair<>(getToolUpgrade(material, "PICKAXE"), 3);
+		} else if (ToolType.AXE.contains(tool)) {
+			return new Pair<>(getToolUpgrade(material, "AXE"), 3);
+		} else if (ToolType.SHOVEL.contains(tool)) {
+			return new Pair<>(getToolUpgrade(material, "SHOVEL"), 1);
+		} else if (ToolType.HOE.contains(tool)) {
+			return new Pair<>(getToolUpgrade(material, "HOE"), 2);
+		} else if (ToolType.HELMET.contains(tool)) {
+			return new Pair<>(getArmorUpgrade(material, "HELMET"), 5);
+		} else if (ToolType.CHESTPLATE.contains(tool)) {
+			return new Pair<>(getArmorUpgrade(material, "CHESTPLATE"), 8);
+		} else if (ToolType.LEGGINGS.contains(tool)) {
+			return new Pair<>(getArmorUpgrade(material, "LEGGINGS"), 7);
+		} else if (ToolType.BOOTS.contains(tool)) {
+			return new Pair<>(getArmorUpgrade(material, "BOOTS"), 4);
+		}
+
+		return null;
+	}
+
+	private static @Nullable Material getToolUpgrade(@NotNull Material material, String tool) {
+		switch (material) {
+			case ACACIA_PLANKS:
+			case BIRCH_PLANKS:
+			case DARK_OAK_PLANKS:
+			case JUNGLE_PLANKS:
+			case OAK_PLANKS:
+			case SPRUCE_PLANKS:
+				return Material.getMaterial("WOODEN_" + tool);
+			case COBBLESTONE:
+				return Material.getMaterial("STONE_" + tool);
+			case IRON_INGOT:
+				return Material.getMaterial("IRON_" + tool);
+			case GOLD_INGOT:
+				return Material.getMaterial("GOLDEN_" + tool);
+			case DIAMOND:
+				return Material.getMaterial("DIAMOND_" + tool);
+			case NETHERITE_INGOT:

```

Comment:
That will unfortunately break in 1.15.2 as NETHERITE_INGOT is not available. Youll need to check for 1.16 first and then check for the material separate

Reply:
Ok, i'll do that